rust/tvm-rt/src/ndarray.rs

```diff
@@ -184,28 +246,19 @@ impl NDArray {
     /// let ctx = Context::cpu(0);
     /// let mut ndarray = NDArray::empty(&mut shape, ctx, DataType::from_str("int32").unwrap());
     /// ndarray.copy_from_buffer(&mut data);
-    /// assert_eq!(ndarray.shape(), Some(&mut shape[..]));
+    /// assert_eq!(ndarray.shape(), shape);
     /// assert_eq!(ndarray.to_vec::<i32>().unwrap(), data);
     /// ```
     pub fn to_vec<T>(&self) -> Result<Vec<T>, NDArrayError> {
```

Comment:
I'm confused about how this function works. If I call it as `my_array.to_vec<u8>`, I'll get back a vector with more elements than if I call it as `my_array.to_vec::<u32>`, which seems like one or the other would not correspond to a flattened version of `self`.

Reply:
Yeah, see above. The problem is that NDArray isn't NDArray<T>. So we trust the type of T given to us by `to_vec::<T>`. That makes it effectively a transmute of the buffer.